veps/sig-storage/virtiofsd-as-infra.md

```diff
@@ -0,0 +1,175 @@
+# VEP #53: Moving virtiofsd to infrastructure for robust live migration
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [X] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+- [ ] (R) Target version is explicitly mentioned and approved
+- [ ] (R) Graduation criteria filled
+
+## Overview
+
+This VEP proposes a fundamental change to how the `virtiofsd` process is managed
+within KubeVirt. Currently, `virtiofsd` runs as an unprivileged process inside a
+dedicated virtiofs container. This setup limits its ability to utilize file
+handles for sharing Persistent Volume Claims (PVCs) with Virtual Machines (VMs)
+and hinders a robust live migration. The proposed solution is to move `virtiofsd`
+as part of the infrastructure. The virtiofs container will remain **rootless**,
+running a dummy process, while the `virt-handler` will launch the `virtiofsd`
+binary and ensure it joins the virtiofs container's namespaces and cgroup.
```

Comment:
I guess the biggest worry about this is going to be CVEs;
We're kind of bypassing the k8s security model by doing this
(though I get this is pretty much required)

Is there anything we could do proactively to ensure we don't hit those?

Reply:
Yeah, we have been burned before messing with the cgroup directly. Since it's not really managed by us, we have been breaking systemds single write rule at the minimum https://systemd.io/CGROUP_DELEGATION/

> Unfortunately I don't have an alternative to suggest at the moment. I wish we could push seitan forward to be adopted by the container engines and then by the k8s sigs, so that we could set capabilities with a limited usage via the pod security API.

Same here. I posted my last take, understanding that there could be inherent limitations that make it impossible. Seitan is really interesting!